plugin/src/main/java/io/jenkins/plugins/analysis/core/steps/WarningChecksPublisher.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    /**
+     * Actually publishes checks to platforms. Afterwards, all warnings are available in corresponding platform's UI.
+     */
+    void publishChecks() {
+        AnalysisResult result = action.getResult();
+        IssuesStatistics totals = result.getTotals();
+
+        StaticAnalysisLabelProvider labelProvider = action.getLabelProvider();
+
+        ChecksPublisher publisher = ChecksPublisherFactory.fromRun(action.getOwner());
+        publisher.publish(new ChecksDetailsBuilder()
+                .withName(labelProvider.getId())
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(extractChecksConclusion(result.getQualityGateStatus()))
+                .withOutput(new ChecksOutputBuilder()
+                        .withTitle(labelProvider.getLinkName())
+                        .withSummary(extractChecksSummary(totals))
+                        .withText(extractChecksText(totals))
+                        .withAnnotations(extractChecksAnnotations(result.getNewIssues()))
+                        .build())
+                .withDetailsURL(action.getAbsoluteUrl())
+                .build());
+    }
+
+    private String extractChecksSummary(final IssuesStatistics statistics) {
+        return String.format("## %d issues in total:\n"
+                        + "- ### %d new issues\n"
+                        + "- ### %d outstanding Issues\n"
+                        + "- ### %d delta issues\n"
+                        + "- ### %d fixed issues",
+                statistics.getTotalSize(), statistics.getNewSize(), statistics.getTotalSize() - statistics.getNewSize(),
+                statistics.getDeltaSize(), statistics.getFixedSize());
+    }
+
+    private String extractChecksText(final IssuesStatistics statistics) {
+        return "## Total Issue Statistics:\n"
+                + generateSeverityText(statistics.getTotalLowSize(), statistics.getTotalNormalSize(),
+                statistics.getTotalHighSize(), statistics.getTotalErrorSize())
+                + "## New Issue Statistics:\n"
+                + generateSeverityText(statistics.getNewLowSize(), statistics.getNewNormalSize(),
+                statistics.getNewHighSize(), statistics.getNewErrorSize())
+                + "## Delta Issue Statistics:\n"
+                + generateSeverityText(statistics.getDeltaLowSize(), statistics.getDeltaNormalSize(),
+                statistics.getDeltaHighSize(), statistics.getDeltaErrorSize());
+    }
+
+    private String generateSeverityText(final int low, final int normal, final int high, final int error) {
+        return "* Error: " + error + "\n"
+                + "* High: " + high + "\n"
+                + "* Normal: " + normal + "\n"
+                + "* Low: " + low + "\n";
+    }
+
+    private ChecksConclusion extractChecksConclusion(final QualityGateStatus status) {
+        switch (status) {
+            case INACTIVE: // TODO:  determine conclusion for it
```

Comment:
Anything else can we do here for `INACTIVE` quality gate?

Reply:
I don't think so for the moment. If users want to fail a GitHub check because of some warnings but do not want to configure a quality gate in Jenkins then they need to open an issue in the warnings plugin.